File: Modules/Common/LES/SIE.Web.LES/MaterialReturnApplys/Commands/MaterialReturnSaveCommand.js
SIE.defineCommand("SIE.Web.LES.MaterialReturnApplys.Commands.MaterialReturnSaveCommand", {
    extend: 'SIE.cmd.FormSave',
    meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    doSave: function (view) {
        var me = this;
        var entity = view.getCurrent();
        if (entity) {
            var childStore = view.findChild("SIE.LES.MaterialReturnApplys.MaterialReturnApplyDetail").getData();
            var data = childStore.data.items;
            for (var i = data.length - 1; i >= 0; i--) {
                data[i].dirty = true;
            }
        }
        var children = view.getChildren();
        var withChildren = children.length > 0;
        view.execute({
            withChildren: withChildren,
            success: function (res) {
                me.onSaved(view, res);
            }
        });
    },
})

File: Modules/Common/LES/SIE.Web.LES/MaterialReturnApplys/Commands/MaterialReturnSaveCommand.cs
using SIE.Domain;
using SIE.LES.MaterialReturnApplys;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.LES.MaterialReturnApplys.Commands
{
    /// <summary>
    /// 退料申请保存命令
    /// </summary>
    public class MaterialReturnSaveCommand : FormSaveCommand
    {
        /// <summary>
        /// 保存前事件
        /// </summary>
        /// <param name="data"></param>
        protected override void OnSaving(Entity data)
        {
            var apply = data as MaterialReturnApply;
            RT.Service.Resolve<MaterialReturnApplyController>().ValidateBeforeSave(apply);
            base.OnSaving(data);
        }
    }
}
